herddb-core/src/main/java/herddb/codec/RecordSerializer.java

```diff
@@ -672,11 +674,11 @@ public static Object deserializePrimaryKey(byte[] key, Table table) {
     }
 
     public static Bytes serializeValue(Map<String, Object> record, Table table) {
-        return new Bytes(serializeValueRaw(record, table));
+        return new Bytes(serializeValueRaw(record, table, 0));
     }
-        
-    public static byte[] serializeValueRaw(Map<String, Object> record, Table table) {
-        ByteArrayOutputStream value = new ByteArrayOutputStream(INITIAL_BUFFER_SIZE);
+
+    public static byte[] serializeValueRaw(Map<String, Object> record, Table table, int expectedSize) {
```

Comment:
I expect this to be another subtle optimization during updates, in fact we will reallocate a buffer of the same size of the previous version of the record, so we want fall into reallocations and copies due to buffer expansions.

In the future for INSERT we could have an average record size per table